lms/djangoapps/courseware/views/views.py

```diff
@@ -499,29 +501,46 @@ def get(self, request, course_id, tab_type, **kwargs):
         course_key = CourseKey.from_string(course_id)
         course = get_course_with_access(request.user, 'load', course_key)
         tab = CourseTabList.get_tab_by_type(course.tabs, tab_type)
-        return super(CourseTabView, self).get(request, course=course, tab=tab, **kwargs)
+        page_context = self.create_page_context(request, course=course, tab=tab, **kwargs)
+        return super(CourseTabView, self).get(request, course=course, page_context=page_context, **kwargs)
 
-    def render_to_fragment(self, request, course=None, tab=None, **kwargs):
+    def create_page_context(self, request, course=None, tab=None, **kwargs):
+        """
+        Creates the context for the fragment's template.
+        """
+        staff_access = has_access(request.user, 'staff', course)
+        supports_preview_menu = tab['supports_preview_menu']
```

Comment:
I know we do this elsewhere, but part me is leery about assuming that every tab has the `supports_preview_menu` field. Maybe turn this into a `get` instead? (Same with `type` below.)